openedx/core/djangoapps/programs/tests/test_views.py

```diff
@@ -0,0 +1,98 @@
+        Client.objects.get_or_create(name="programs", client_type=0)
+        self.user = UserFactory()
+        self.programs_api_response = {
+            "results": [
+                {
+                    'category': 'xseries',
+                    'status': 'active',
+                    'subtitle': 'Dummy program for testing',
+                    'name': 'First Program',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization', 'key': 'edX'},
+                            'display_name': 'Demo Course',
+                            'key': 'TEST_A',
+                            'run_modes': [{'sku': '', 'mode_slug': 'ABC', 'course_key': 'edX/DemoX/Run'}]
+                        }
+                    ]
+                }
+            ]
+        }
+
+    def test_get_course_programs_with_valid_user_and_courses(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        the expected format.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            mock_get.return_value = self.programs_api_response
+            programs = get_course_programs_for_dashboard(self.user, ['edX/DemoX/Run', 'valid/edX/Course'])
+            expected_output = {
+                'edX/DemoX/Run': {
+                    'category': 'xseries',
+                    'status': 'active',
+                    'course_codes': [
+                        {
+                            'organization': {'display_name': 'Test Organization', 'key': 'edX'},
+                            'display_name': 'Demo Course',
+                            'key': 'TEST_A',
+                            'run_modes': [{'sku': '', 'mode_slug': 'ABC', 'course_key': 'edX/DemoX/Run'}]
+                        }
+                    ],
+                    'subtitle': 'Dummy program for testing',
+                    'name': 'First Program'
+                }
+            }
+            self.assertEqual(expected_output, programs)
+
+    def test_get_course_programs_with_non_existing_courses(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        only those program courses which exists in the programs api response.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            mock_get.return_value = self.programs_api_response
+            self.assertEqual(
+                get_course_programs_for_dashboard(self.user, ['invalid/edX/Course']), {}
+            )
+
+    def test_get_course_programs_with_invalid_user(self):
+        """ Test that the method 'get_course_programs_for_dashboard' returns
+        None for an anonymous user.
+        """
+        # mock the request call
+        with patch('slumber.Resource.get') as mock_get:
+            mock_get.return_value = self.programs_api_response
```

Comment:
If we ever called the API on behalf of an invalid or anonymous user, the response from the API would be a 401.  Also, I don't think it's important to test this function with an anonymous user, since they would never be looking at a student dashboard - is the reason for this test to ensure the exception handling is working?